### DISCUSSION TITLE
basic caddy configuration

### DISCUSSION THREAD

**Discussion by heapxor** [nextcloud/all-in-one#998] [Questions] [answered]
hello,
is there some basic Caddy configuration that can run on top of the nextcloud and expose it via port 80 via http?

the reason why i am asking is that i have other nginx box acting as reverse proxy - that will add ssl/tls on top of the nexcloud..

@szaimen assuming not everyone wants to use built in caddy reverse proxy capabilities... assume that there is already reverse proxy nginx in the setup that has to be reused for that.

thanks!

**Comment by Zoey2936** [comment]
Did you read this?
https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md

**Reply by Zoey2936** [reply]
If you set the Apache port env, while starting the mastercontainer, you can in principle run the nextcloud behind every reverse proxy

**Reply by heapxor** [reply]
yes i did ... but as i said i have different requirements ie i need to run my own reverse proxy ,... 

i dont know why apache port env.... it wont be possible with caddy? I was asking why there are no default conf available for apache/caddy/nginx ... running as web server only.

**Reply by Zoey2936** [reply]
What do you exactly mean with "my own reverse proxy"?
If you want to use for example nginx as reverse proxy than just follow: https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md

**Reply by heapxor** [reply]
separate machine running nginx acting as reverse proxy.

problem as was stated - i do need default caddy/apache web server configs for nextcloud.

**Reply by Zoey2936** [reply]
So you have one server running nextcloud-aio and one with a reverse proxy, and now you want to expose a http port of nextcloud aio, which is then proxied by your reverse proxy?

**Reply by heapxor** [reply]
yes exactly.

**Reply by Zoey2936** [reply]
And what is your docker-compose file / docker run command you use to start the master container?

**Reply by heapxor** [reply]
i did slightly different setup. 
thats why i was asking if there are caddy/apache default confs.

at the moment i am getting 
Internal Server Error

The server encountered an internal error and was unable to complete your request.
Please contact the server administrator if this error reappears multiple times, please include the technical details below in your report.
More details can be found in the webserver log.

**Reply by Zoey2936** [reply]
What do you mean exactly with "slightly different"?
If you just want to run it behind a reverse proxy, add the `APACHE_PORT` env to the docker run command / docker-compose file and set up your reverse proxy to proxy to this port using http?
Or did you used docker rootless / manual install?

**Reply by heapxor** [reply]
and on the posted link they says...

Nginx-Proxy
click here to expand

Unfortunately it is not possible to configure nginx-proxy in a way that works because it completely relies on environmental variables of the docker containers itself. Providing these variables does not work as stated above.

If you really want to use AIO, we recommend you to switch to caddy. It is simply amazing!
Of course understandable if that is not possible for you.

Apart from that, there is this: [manual-install](https://github.com/nextcloud/all-in-one/tree/main/manual-install)

So i means it will not work? this is extra confusing.

**Reply by heapxor** [reply]
> What do you mean exactly with "slightly different"?
> If you just want to run it behind a reverse proxy, add the `APACHE_PORT` env to the docker run command / docker-compose file and set up your reverse proxy to proxy to this port using http?
> Or did you used docker rootless / manual install?

the root question here was... is there some caddy/apache default web conf ...

**Reply by Zoey2936** [reply]
Do you mean these?
https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md#caddy-recommended 
https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md#apache

**Reply by heapxor** [reply]
No, 
i dont understand that - do i write in some non readable language?


I stated the problem clearly;
1) run nextcloud (on host A, or in docker) with default apaache / caddy conf where its web interface is exposed via port 80 

2) run on separate box (host B) nginx server which will act as reverse proxy and connect to 1)

what is not clear here?

**Reply by Zoey2936** [reply]
> No, i dont understand that - do i write in some non readable language?
> 
> I stated the problem clearly;
> 
>     1. run nextcloud (on host A, or in docker) with default apaache / caddy conf where its web interface is exposed via port 80
> 
>     2. run on separate box (host B) nginx server which will act as reverse proxy and connect to 1)
> 
> 
> what is not clear here?

Why you do it so complicated, just start the mastercontainer with the env `APACHE_PORT=80` and on your nginx server just use this [config](https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md#nginx) and replace `localhost:11000` with `ip-of-host-a:80`

**Reply by heapxor** [reply]
i think its not important why i am doing it so complicated.

i cant do it as you are suggesting and i was asking if there is some default CADDY/APACHE web conf.

at the moment my caddy instance running on top of the nextcloud on port 80 returns>
**,"Status":["500 Internal Server Error"]**

caddy conf>

`
{
        # debug
        #acme_ca https://acme-staging-v02.api.letsencrypt.org/directory
        email email@pm.me
        default_sni clode.duckdns.o
}

:80 {
        root * /usr/local/www/nextcloud
        file_server
        log {
                output file /var/log/clode.duckdns.org.log
        }

        php_fastcgi 127.0.0.1:9000 {
                env front_controller_active true
        }

        #tls {
        #       x
        #}

        header {
                # enable HSTS
                # Strict-Transport-Security max-age=31536000;
        }

        # client support (e.g. os x calendar / contacts)
        redir /.well-known/carddav /remote.php/dav 301
        redir /.well-known/caldav /remote.php/dav 301
        redir /.well-known/webfinger /index.php/.well-known/webfinger 301
        redir /.well-known/nodeinfo /index.php/.well-known/nodeinfo 301

       # .htaccess / data / config / ... shouldn't be accessible from outside
        @forbidden {
                path /.htaccess
                path /data/*
                path /config/*
                path /db_structure
                path /.xml
                path /README
                path /3rdparty/*
                path /lib/*
                path /templates/*
                path /occ
                path /console.php
        }

        respond @forbidden 404
}
`
anyone any idea? thanks

**Reply by heapxor** [reply]
i replaced caddy with apache and all works as expected.

**Comment by szaimen** [comment, marked as answer]
Hi, I still do not understand what you are trying to achieve in detail (as it looks like you are trying to use some components of AIO and some of somewhere else?) but glad that switching to Apache solved it!
